argo_jupyter_scheduler/utils.py

```diff
@@ -65,6 +72,12 @@ def authenticate():
     return global_config
 
 
+def gen_timestamp(time: int):
+    # See create_output_filename in jupyter-scheduler
+    timestamp = datetime.fromtimestamp(time / 1e3)  # noqa: DTZ006
+    return timestamp.strftime("%Y-%m-%d-%I-%M-%S-%p")
```

Comment:
This is the important part to make files downloadable via the web UI. They need to be in the correct directory and have this exact format, which is determined in the original jupyter-scheduler extension as follows:

```py
def create_output_filename(input_filename: str, create_time: int, output_format: str = None) -> str:
    """Creates output filename from input_filename, create_time and output_format"""
    basefilename = os.path.splitext(input_filename)[0]
    timestamp = datetime.fromtimestamp(create_time / 1e3).strftime("%Y-%m-%d-%I-%M-%S-%p")
    if output_format:
        return f"{basefilename}-{timestamp}.{output_format}"
    else:
        return f"{basefilename}-{timestamp}"
```

And the look up in the original extension is happening here:

```
class JobFilesManager:
    scheduler = None

    def __init__(self, scheduler: Type[BaseScheduler]):
        self.scheduler = scheduler

    async def copy_from_staging(self, job_id: str, redownload: Optional[bool] = False):
        job = await ensure_async(self.scheduler.get_job(job_id, False))
        staging_paths = await ensure_async(self.scheduler.get_staging_paths(job))
        print("XXX: ", staging_paths)
        output_filenames = self.scheduler.get_job_filenames(job)
        output_dir = self.scheduler.get_local_output_path(job)

        p = Process(
            target=Downloader(
                output_formats=job.output_formats,
                output_filenames=output_filenames,
                staging_paths=staging_paths,
                output_dir=output_dir,
                redownload=redownload,
            ).download
        )
        p.start()
```

Here is an example value printed by a patched jupyter-scheduler extension when you attempt to download files:

<img width="858" alt="Screen Shot 2023-12-08 at 02 48 23" src="https://github.com/nebari-dev/argo-jupyter-scheduler/assets/10469885/ac054cd1-e180-4167-87ba-315486529d1e">

So we need to create files having this exact same format or the look up will fail and you won't be able to download anything. The timestamp value must match the `start_time` of the job you're trying to download files from.

Reply:
Theoretically, we could try reusing `create_output_filename` from the original extension instead of having a copy here. I think it should work. I created a separate timestamp function when I was attempting the splicing idea, which required me to use strings. But I have no time right now to make a change where we just use `create_output_filename`.

Reply:
Realized it's an easy fix to do, so made the change.